### PR TITLE
Add linear slack method

### PR DESCRIPTION
**Pull request by rahmani15**
Add `linear` option as a new `slack_method` for  `DiscreteQuadraticModel.add_linear_inequality_constraint` method. This will add only one discrete variable to an inequality constraint with linear number of cases, according to right hand side value of the constraint.

### CODE REVIEW

**Comment by arcondello** [comment]
Can you add a release note?

**Comment by rahmani15** [comment]
It is there. I added that.

**Comment by arcondello** [comment]
oops, missed that somehow

**Review by arcondello — APPROVED** [review]

**Comment by rahmani15** [comment]
my release has a typo though, will fix it once I get all comments.

**Comment by arcondello** [comment]
Code looks good, go ahead and fix the typo and then we can merge.

File: releasenotes/notes/add-linear-slack-dqm-25312a5fe23574e6.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+features:
+  - |
+    Add `linear` option as a new `slack_method` for
```
Comment (arcondello) [inline releasenotes/notes/add-linear-slack-dqm-25312a5fe23574e6.yaml:4]: Need ` `` ` rather than ``` ` ``` for the code snippets in RST

Comment (rahmani15) [reply releasenotes/notes/add-linear-slack-dqm-25312a5fe23574e6.yaml]: this is a science :), I fixed that.

Comment (arcondello) [reply releasenotes/notes/add-linear-slack-dqm-25312a5fe23574e6.yaml]: You missed the ones around `DiscreteQuadraticModel.add_linear_inequality_constraint`

**Pull request merged**